src/context/PortfolioProvider/PortfolioContext.tsx

```diff
@@ -242,9 +227,22 @@ export const PortfolioProvider = ({ children }: { children: React.ReactNode }) =
     if (!tx) return
 
     if (tx.caip2 === cosmosChainId) {
-      // @TODO: Remove this once stakingData slice is refactored into account data
-      refetchStakingDataByTxId(txId)
-
+      // TODO: this probably shouldn't belong here, otherwise it will only work after first websocket Tx
+      // Is there a better place to move this in so we can load validators right after portfolio accounts are available?
+      const validators =
+        // TODO: That's a hacky hack during dev, remove me obviously - not a clean nor reliable way to get accountSpecifier
+        portfolioAccounts[`cosmos:cosmoshub-4:${tx.address}`]?.validatorIds
```

Comment:
we have a function called something like makeUniqueTxId that is a pure function that takes a full `Tx` and returns the id